sklearn/neural_network/tests/test_mlp.py

```diff
@@ -479,6 +479,21 @@ def test_verbose_sgd():
     assert 'Iteration' in output.getvalue()
 
 
+def test_verbose_lbfgs():
+    # Test verbose.
+    X = [[3, 2], [1, 6]]
+    y = [1, 0]
+    clf = MLPClassifier(algorithm='l-bfgs', max_iter=2, verbose=10,
```

Comment:
you could avoid duplication with something like 
```python
for algorithm in ['l-bfgs', 'sgd']:
    clf = MLPClassifier(algorithm=algorithm, max_iter=2, verbose=10,
                        hidden_layer_sizes=2)
```
But you can keep the try/finally structure, with is better than in `test_verbose_sgd`.